File: cat.zig
const std = @import("std");

pub fn main() !void {
    const gpa = std.heap.page_allocator;
    const stdout = std.io.getStdOut().writer();
    const stdin = std.io.getStdIn().reader();

    var buffer: [4096]u8 = undefined;
    while (true) {
        const n = try stdin.read(&buffer);
        if (n == 0) break;
        try stdout.writeAll(buffer[0..n]);
    }
}

File: cat.rs
use std::io::{self, Read, Write};

fn main() -> io::Result<()> {
    let mut stdin = io::stdin();
    let mut stdout = io::stdout();
    let mut buffer = [0u8; 4096];

    loop {
        let n = stdin.read(&mut buffer)?;
        if n == 0 { break; }
        stdout.write_all(&buffer[..n])?;
    }
    Ok(())
}
